File: src/main/java/com/ii/app/services/interfaces/TransactionService.java
package com.ii.app.services.interfaces;

import com.ii.app.dto.in.TransactionIn;
import com.ii.app.dto.out.TransactionOut;
import com.ii.app.models.Transaction;

import java.util.List;

public interface TransactionService {
    TransactionOut create(TransactionIn transactionDTO);

    List<TransactionOut> findAll();

    List<TransactionOut> findAllByBankAccountId(Long bankAccountId);
}

File: front/src/app/main/services/transaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Transaction } from '../models/transaction';
import { TransactionIn } from '../models/history-element';
import {environment} from "src/environments/environment";

@Injectable({
  providedIn: 'root'
})
export class TransactionService {

  url = environment.apiUrl + '/api/transaction';

  constructor(private http: HttpClient) { }

  create(transaction: string) {
    return this.http.post<Transaction>(this.url, transaction);
  }

  findAllByBankAccountId(bankAccountId: number){
    return this.http.get<TransactionIn[]>(this.url + '/byAccount/' + bankAccountId);
  }
}
